x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/TrainedModelAssignmentNodeService.java

```diff
@@ -184,6 +184,7 @@ void stop() {
 
     void loadQueuedModels(ActionListener<Boolean> rescheduleImmediately) {
         if (stopped) {
+            rescheduleImmediately.onResponse(false);
```

Comment:
This is to not leave the listener dangling - the `if (stopped)` check happens again from the schedule function so it won't kick off a new execution